nextcord/application_command.py

```diff
@@ -947,11 +992,8 @@ async def invoke_error(self, interaction: Interaction, error: Exception) -> None
         """|coro|
         Invokes the error handler if available.
         """
-        if self.has_error_handler() and self.error_callback is not None:
-            if self.parent_cog:
-                await self.error_callback(self.parent_cog, interaction, error)
-            else:
-                await self.error_callback(interaction, error)
+        if self.has_error_handler():
```

Comment:
At that point, why even have the function to check it? I mean, if we never use it.

Reply:
Useful shorthand for users? Just because it's defined doesn't mean it has to be used by the underlying internal code.

Reply:
I don't think this is a good idea to change. That would mean there is not one spot to change if it ever needs a change, but multiple.
And that, at least for me, is a bigger issue then pyright not being able to put a function with one line into context.

Reply:
> I don't think this is a good idea to change. That would mean there is not one spot to change if it ever needs a change, but multiple.

`has_error_callback` is currently only referenced once in `application_command.py`. With this suggestion, if error callbacks were changed in some way (maybe to allow multiple or something), it would be a small change. It's really not a big deal in this case.

If you're still not convinced, you can alternatively add this under the if statement:
```python
if TYPE_CHECKING:
    error_callback = cast(Callable, self.error_callback)

await error_callback(interaction, error)
```